Mathlib/RingTheory/TwoSidedIdeal/Operations.lean

```diff
@@ -0,0 +1,199 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import Mathlib.RingTheory.TwoSidedIdeal.Lattice
+import Mathlib.RingTheory.TwoSidedIdeal.BigOperators
+import Mathlib.RingTheory.Congruence.Opposite
+import Mathlib.Algebra.BigOperators.Ring
+import Mathlib.Data.Fintype.BigOperators
+import Mathlib.RingTheory.Ideal.Basic
+import Mathlib.Order.GaloisConnection
+
+
+/-!
+# Operations on Two-Sided-Ideals
+
+This file defines operations on two-sided-ideals of a ring `R`.
+
+## Main definitions and results
+- `TwoSidedIdeal.span`: the span of `s ⊆ R` is the smallest two-sided-ideal containing the set.
+- `TwoSidedIdeal.mem_span_iff_exists`: an element `x` is in the span of `s` if and only if it can be
+  written as a finite sum of the form `∑ i, xL i * y i * xR i` where `xL i ∈ s`, `y i ∈ R`, and
+  `xR i ∈ s`.
+
+- `TwoSidedIdeal.comap`: pullback of a two-sided-ideal; defined as the preimage of a
+  two-sided-ideal.
+- `TwoSidedIdeal.map`: pushout of a two-sided-ideal; defined as the span of the image of a
```

Comment:
I don't think pushout is what you mean here. Perhaps "push forward"?

Reply:
indeed, thanks!